core/src/test/java/org/apache/brooklyn/core/test/entity/TestSizeRecordingClusterImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.core.test.entity;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.brooklyn.entity.group.DynamicClusterImpl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class TestSizeRecordingClusterImpl extends DynamicClusterImpl implements TestSizeRecordingCluster {
```

Comment:
Can we instead just use the existing `org.apache.brooklyn.core.test.entity.TestCluster`?

In your test, you're configuring `TestCluster.MEMBER_SPEC` so it calling `super.resize()` should be fine I'd have thought.